services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationService.kt

```diff
@@ -316,21 +319,82 @@ open class ApplicationService(
         }
         applicationHistory.events
             .sortedBy { it.eventTime }
-            .forEach { event ->
-                application.alluStatus = event.newStatus
-                application.applicationIdentifier = event.applicationIdentifier
-                logger.info {
-                    "Updating application with new status, " +
-                        "id=${application.id}, " +
-                        "alluid=${application.alluid}, " +
-                        "application identifier=${application.applicationIdentifier}, " +
-                        "new status=${application.alluStatus}, " +
-                        "event time=${event.eventTime}"
-                }
-            }
+            .forEach { handleApplicationEvent(application, it) }
         applicationRepository.save(application)
     }
 
+    private fun handleApplicationEvent(
+        application: ApplicationEntity,
+        event: ApplicationStatusEvent
+    ) {
+        application.alluStatus = event.newStatus
+        application.applicationIdentifier = event.applicationIdentifier
+        logger.info {
+            "Updating application with new status, " +
+                "id=${application.id}, " +
+                "alluid=${application.alluid}, " +
+                "application identifier=${application.applicationIdentifier}, " +
+                "new status=${application.alluStatus}, " +
+                "event time=${event.eventTime}"
+        }
+        if (event.newStatus == ApplicationStatus.DECISION) {
+            sendDecisionReadyEmails(application, event.applicationIdentifier)
+        }
+    }
+
+    private fun sendDecisionReadyEmails(
+        application: ApplicationEntity,
+        applicationIdentifier: String
+    ) {
+        val receivers =
+            application.applicationData
+                .customersWithContacts()
+                .flatMap { it.contacts }
+                .filter { it.orderer }
+
+        if (receivers.isEmpty()) {
+            logger.error {
+                "No receivers found for decision ready email, not sending any." +
+                    "applicationId=${application.id}, applicationIdentifier=${applicationIdentifier}"
+            }
```

Comment:
Yes, I thought it was better to do the next check as well since `receivers.forEach` doesn't do anything when `receivers` is null anyway. But it's better to return early so that code changes don't cause surprises.